src/main/java/org/opentripplanner/middleware/controllers/api/TripHistoryController.java

```diff
@@ -0,0 +1,137 @@
+                limit = Integer.parseInt(paramLimit);
+                if (limit <= 0) {
+                    limit = DEFAULT_LIMIT;
+                }
+            }
+        } catch (NumberFormatException e) {
+            LOG.error("Unable to parse {} value of {}. Using default limit: {}", LIMIT_PARAM_NAME,
+                paramLimit, DEFAULT_LIMIT, e);
+        }
+
+        String paramFromDate = HttpUtils.getParamFromRequest(request, FROM_DATE_PARAM_NAME, true);
+        Date fromDate = getDate(request, FROM_DATE_PARAM_NAME, paramFromDate, LocalTime.MIDNIGHT);
+
+        String paramToDate = HttpUtils.getParamFromRequest(request, TO_DATE_PARAM_NAME, true);
+        Date toDate = getDate(request, TO_DATE_PARAM_NAME, paramToDate, LocalTime.MAX);
+
+        if (fromDate != null && toDate != null && toDate.before(fromDate)) {
+            logMessageAndHalt(request, HttpStatus.BAD_REQUEST_400,
+                String.format("%s (%s) before %s (%s)", TO_DATE_PARAM_NAME , paramToDate, FROM_DATE_PARAM_NAME,
+                    paramFromDate));
+        }
+
+        Bson filter = buildFilter(userId, fromDate, toDate);
+        return JsonUtils.toJson(tripRequest.getFilteredWithLimit(filter, limit));
+    }
+
+    /**
+     * Build the filter which is passed to Mongo based on available parameters
+     */
+    private static Bson buildFilter(String userId, Date fromDate, Date toDate) {
+
+        Set<Bson> clauses = new HashSet<>();
+
+        // user id is required, so as a minimum return all trip requests for user
+        clauses.add(eq("userId", userId));
+
+        // Get all trip requests that occurred from supplied start date.
+        if (fromDate != null) {
+            clauses.add(gte("dateCreated", fromDate));
+        }
+
+        // Get all trip requests that occurred until the supplied end date.
+        if (toDate != null) {
+            clauses.add(lte("dateCreated", toDate));
+        }
+
+        return Filters.and(clauses);
+    }
+
+    /**
+     * Get date from request parameter and convert to java.util.Date at a specific time of day
+     */
+    private static Date getDate(Request request, String paramName, String paramValue, LocalTime timeOfDay) {
+
+        // no date value to work with
+        if (paramValue == null) {
+            return null;
+        }
+
+        String expectedDatePattern = "yyyy-MM-dd";
+        LocalDate localDate = null;
+        try {
+            localDate = DateUtils.getDateFromParam(paramName, paramValue, expectedDatePattern);
+        } catch (DateTimeParseException e) {
+            logMessageAndHalt(request, HttpStatus.BAD_REQUEST_400,
+                String.format("%s value: %s is not a valid date. Must be in the format: %s", paramName, paramValue,
+                    expectedDatePattern));
+        }
+
+        if (localDate == null) {
+            return null;
+        }
+
+        return Date.from(localDate.atTime(timeOfDay)
+                .atZone(ZoneId.systemDefault())
```

Comment:
This method should note the use of the system timezone for parsing a date.

Reply:
Updated.